spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/BasicCassandraPredicatePushDown.scala

```diff
@@ -87,40 +87,75 @@ class BasicCassandraPredicatePushDown[Predicate : PredicateOps](
 
 
   /**
-   * Selects partition key predicates for pushdown:
-   * 1. Partition key predicates must be equality or IN predicates.
-   * 2. Only the last partition key column predicate can be an IN.
-   * 3. All partition key predicates must be used or none.
+    * Selects partition key predicates for pushdown:
+    * 1. Partition key predicates must be equality or IN predicates.
+    * 2. Prior to Cassandra Version 2.2 (ProtocolVersion V4), only the last partition key column predicate can be an IN.
+    * In 2.2 you can use the IN operator on any partition key column
+    * 3. All partition key predicates must be used or none.
    */
   private val partitionKeyPredicatesToPushDown: Set[Predicate] = {
-    val (eqColumns, otherColumns) = partitionKeyColumns.span(eqPredicatesByName.contains)
-    val inColumns = otherColumns.headOption.toSeq.filter(inPredicatesByName.contains)
+    val (eqColumns, inColumns) = pv match {
+      case pv if pv >= V4 => {
```

Comment:
this is just a hidden `if`

Reply:
Are you referring to the 'shadowing issue by a variable' as the case statement uses same name (pv) as initial value(pv). Have fixed this in recent pull request.